src/debugAdapter/goDebug.ts

```diff
@@ -1587,6 +1585,75 @@ class GoDebugSession extends LoggingDebugSession {
 		});
 	}
 
+	private logDelveError(err: any, message: string) {
+		if (err) {
+			let errorMessage = err.toString();
+			// Handle unpropagated fatalpanic errors with a more user friendly message:
+			// https://github.com/microsoft/vscode-go/issues/1903#issuecomment-460126884
+			// https://github.com/go-delve/delve/issues/852
+			// This affects macOS only although we're agnostic of the OS at this stage, only handle the error
+			if (errorMessage === 'bad access') {
+				errorMessage = 'unpropagated fatalpanic: signal SIGSEGV (EXC_BAD_ACCESS). This fatalpanic is not traceable on macOS, see https://github.com/go-delve/delve/issues/852';
+			}
+
+			logError(message + ' - ' + errorMessage);
+
+			if (err.toString() === 'bad access') {
```

Comment:
The same `if` check exists a few lines above, so consider moving this log statement to the previous if block

Reply:
I don't like the duplicate if either but I did it so the warning comes right before the stacktrace and not before the errorMessage:

Current:
<img width="839" alt="Screen Shot 2019-12-17 at 11 51 25" src="https://user-images.githubusercontent.com/6326271/71006116-9ba1b880-20c3-11ea-948d-7073f021aa97.png">

Simplified:
<img width="838" alt="Screen Shot 2019-12-17 at 11 49 13" src="https://user-images.githubusercontent.com/6326271/71006111-99d7f500-20c3-11ea-9239-aefb86581a60.png">

I don't think it is that much of an issue so if you're fine with that I can go on with the single if version

Reply:
Understood, in that case let's keep the current order